web3.js/test/connection.test.ts

```diff
@@ -2002,6 +2002,27 @@ describe('Connection', () => {
       await connection.removeRootChangeListener(subscriptionId);
     });
 
+    it('logs notification', async () => {
+      let listener: number | undefined;
+      const owner = new Account();
+      const [logsRes, ctx] = await new Promise(resolve => {
+        listener = connection.onLogs('all', (logs, ctx) => {
+          resolve([logs, ctx]);
+        });
+        // Execute a transaction so that we can pickup its logs.
+        connection.requestAirdrop(owner.publicKey, 1);
+      });
+      expect(ctx.slot).to.be.greaterThan(0);
+      expect(logsRes.logs.length).to.eq(2);
+      expect(logsRes.logs[0]).to.eq(
+        'Program 11111111111111111111111111111111 invoke [1]',
+      );
+      expect(logsRes.logs[1]).to.eq(
+        'Program 11111111111111111111111111111111 success',
+      );
+      await connection.removeOnLogsListener(listener!);
```

Comment:
what is that bang for?

Reply:
To tell TypeScript `listener` is not undefined, though it's not necessary it seems.

Reply:
That's super useful, thanks for the tip!

Reply:
Oops I was incorrect. It is indeed necessary, here.